Documentation/common-problems.md

```diff
@@ -23,25 +26,149 @@ One of the first things that should be done is to start the [rook-tools pod](./t
+```
+From the output above, we can see that the agents were not able to bind mount to `/usr/libexec/kubernetes` on the host they are scheduled to run on.
+For some environments, this default path is read-only and therefore a better path must be provided to the agents.
+
+First, clean up the agents deployment with:
+```console
+kubectl -n rook-system delete daemonset rook-agent
+```
+Once the `rook-agent` pods are gone, **follow the instructions in the [Flexvolume configuration pre-reqs](./k8s-pre-reqs.md#flexvolume-configuration)** to ensure a good value for `--volume-plugin-dir` has been provided to the Kubelet.
+After that has been configured, and the Kubelet has been restarted, start the agent pods up again by restarting `rook-operator`:
+```console
+kubectl -n rook-system delete pod `kubectl -n rook-system get pod -l app=rook-operator -o jsonpath='{.items[*].metadata.name}'`
+```
+
+### Kubelet Restart
+#### **Kubernetes 1.7.x and earlier only**
+If the `rook-agent` pods are all in the `Running` state then another thing to confirm is that **if you are running on Kubernetes 1.7.x or earlier**, the Kubelet must be restarted after the `rook-agent` pods are running.
+
+A symptom of this can be found in the Kubelet's log/journal, with the following error saying `no volume plugin matched`:
+```console
+Oct 30 22:23:03 core-02 kubelet-wrapper[31926]: E1030 22:23:03.524159   31926 desired_state_of_world_populator.go:285] Failed to add volume "mysql-persistent-storage" (specName: "pvc-9f273fbc") for pod "9f2ff89a-bdbf" to desiredStateOfWorld. err=failed to get Plugin from volumeSpec for volume "pvc-9f273fbc" err=no volume plugin matched
+```
+If you encounter this, just **restart the Kubelet process**, as described in the **[Restart Kubelet](./kubernetes.md#deploy-rook)** section of the Rook deployment guide.
+
+### Volume Creation
+The volume must first be created in the Rook cluster and then bound to a volume claim before it can be mounted to a pod.
+Let's confirm that with the following commands and their output:
+```console
+> kubectl get pv
+NAME                                       CAPACITY   ACCESSMODES   RECLAIMPOLICY   STATUS     CLAIM                    STORAGECLASS   REASON    AGE
+pvc-9f273fbc-bdbf-11e7-bc4c-001c428b9fc8   20Gi       RWO           Delete          Bound      default/mysql-pv-claim   rook-block               25m
+
+> kubectl get pvc
+NAME             STATUS    VOLUME                                     CAPACITY   ACCESSMODES   STORAGECLASS   AGE
+mysql-pv-claim   Bound     pvc-9f273fbc-bdbf-11e7-bc4c-001c428b9fc8   20Gi       RWO           rook-block     25m
+```
+Both your volume and its claim should be in the `Bound` status.
+If one or neither of them is not in the `Bound` status, then look for details of the issue in the `rook-operator` logs:
+```console
+kubectl -n rook-system logs `kubectl -n rook-system -l app=rook-operator get pods -o jsonpath='{.items[*].metadata.name}'`
```

Comment:
i think you can simplify this one as well with `kubectl -n rook-system logs -l app=rook-operator`

Reply:
i thought so too, but when you do `kubectl logs` with a label, it treats it like a tail and only gives you the last 15 lines or so!  i couldn't figure a way around that besides giving it the full pod name.

Reply:
oh that's odd